### PR TITLE
fix bug when using deprecated password field

### PR DESCRIPTION
**Pull request by raymonstah**
# Overview

Fixes what appears to be a bug when supplying the `password` field.

### CODE REVIEW

**Review by fairclothjm — APPROVED** [review]

File: redis_elasticache_client.go [hunk 1]
```diff
@@ -48,13 +48,13 @@ func (r *redisElastiCacheDB) Initialize(_ context.Context, req dbplugin.Initiali
 		accessKey = r.config.Username
 	}
 	secretKey := r.config.SecretAccessKey
-	if secretKey == "" && r.config.SecretAccessKey != "" {
-		secretKey = r.config.SecretAccessKey
+	if secretKey == "" && r.config.Password != "" {
+		secretKey = r.config.Password
```
Comment (raymonstah) [inline redis_elasticache_client.go:52]: @maxcoulombe just wanted to confirm this was the intention before I merge.
Thanks!

Comment (maxcoulombe) [reply redis_elasticache_client.go:52]: Whoops, this is definitely a mistake.

This looks good to me. To be sure, in plain english:
* If the secretKey field is provided it has precedence
* If not, we try to look for the deprecated password field
* If both are unspecified we leave it blank so the util library fallbacks on the default provider chain e.g. try to load the creds from env vars

**Review by maxcoulombe — APPROVED** [review]
